docs/zkapps/tutorials/04-in-browser-react-ui.mdx

```diff
@@ -0,0 +1,417 @@
+114
+115     console.log('creating proof...');
+116     await state.zkappWorkerClient!.proveUpdateTransaction();
+117
+118     const transactionHash = await state.zkappWorkerClient!.sendUpdateTransaction();
+119
+120     console.log(
+121       'See transaction at https://berkeley.minaexplorer.com/transaction/' + transactionHash
+122     );
+123
+124     setState({ ...state, creatingTransaction: false });
+125   }
+126
+127   // -------------------------------------------------------ts
+...
+```
+
+And second, a function that will get the latest zkApp state:
+
+```ts
+...
+127   // -------------------------------------------------------
+128   // Refresh the current state
+129
+130   const onRefreshCurrentNum = async () => {
+131     console.log('getting zkApp state...');
+132     await state.zkappWorkerClient!.fetchAccount({ publicKey: state.zkappPublicKey! })
+133     const currentNum = await state.zkappWorkerClient!.getNum();
+134     console.log('current state:', currentNum.toString());
+135
+136     setState({ ...state, currentNum });
+137   }
+138
+139   // -------------------------------------------------------ts
+...
+```
+
+Lastly, we will update the `return <div/>` placeholder we originally inserted, with a ui to show the user the state of our application:
+
+```ts
+...
+139   // -------------------------------------------------------
+140   // Create UI elements
+141
+142   let setupText = state.hasBeenSetup ? 'SnarkyJS Ready' : 'Setting up SnarkyJS...';
+143   let setup = <div> { setupText } </div>
+144
+145   let accountDoesNotExist;
+146   if (state.hasBeenSetup && !state.accountExists) {
+147     const faucetLink = "https://faucet.minaprotocol.com/?address=" + state.publicKey!.toBase58();
+148     accountDoesNotExist = <div>
+149       Account does not exist. Please visit the faucet to fund this account
+150       <a href={faucetLink} target="_blank" rel="noreferrer"> [Link] </a>
+151     </div>
+152   }
+153
+154   let mainContent;
+155   if (state.hasBeenSetup && state.accountExists) {
+156     mainContent = <div>
+157       <button onClick={onSendTransaction} disabled={state.creatingTransaction}> Send Transaction </button>
+158       <div> Current Number in zkApp: { state.currentNum!.toString() } </div>
+159       <button onClick={onRefreshCurrentNum}> Get Latest State </button>
+160     </div>
+161   }
+162
+163   return <div>
+164    { setup }
+165    { accountDoesNotExist }
+166    { mainContent }
+167   </div>
+168 }
+```
+
+We divide our UI into 3 sections:
+
+* `setup` lets the user know when the zkapp has finished loading.
+* `accountDoesNotExist` gives the user a link to the faucet if their account hasn't been funded.
+* `mainContent` shows the current state of the zkApp, and buttons to interact with it.
+
+The buttons allow the user to create a transaction, or refresh the current state of the application, by triggering the `onSendTransaction` and `onRefreshCurrentNum` functions we wrote above.
```

Comment:
`onSendTransaction()` and `onRefreshCurrentNum()`

Would read more clearly (instantly recognizable as functions)